### DISCUSSION TITLE
Splitting a Face, while preserving singular solid

### DISCUSSION THREAD

**Discussion by fsfeller** [CadQuery/cadquery#1674] [Q&A]
Hi,

To make the workflow with Ansys as seamless as possible, I need the cad file already prepared to, for example, have the sections of an object with different properties as a seperate face of a solid. This way each face can be selected easily and manipulated individually.

As an example case, consider i have a simple cylinder. Now I want to split its curved surfaces to have two seperate surfaces, while still only having one solid body.
I cannot manage to achieve this and would really appreaciate, if someone has an idea.

Thank you very much!

**Comment by adam-urbanczyk** [comment]
Is this what you need: https://github.com/CadQuery/cadquery/pull/1353 ?

**Comment by pruj-speed** [comment]
@adam-urbanczyk  I believe the ask is to split a face using a polygon. This is typical in simulations to select surfaces and apply a boundary condition on them.

Given a cube and a triangle shape, one would like to see that rectangular face of the cube be split into two face, one a triangle and other the boolean of the rectangle and triangle. See image for illustration. Turns out I would also need this functionality for my carpet on a staircase like CAD im trying to generate https://github.com/CadQuery/cadquery/discussions/1698 .

![image](https://github.com/user-attachments/assets/5370db67-3b48-4d3b-beec-8117a8b16cfc)

![image](https://github.com/user-attachments/assets/75d409ea-02b4-4e87-adbd-95eb3d0e654a)

I tried to implement it using a script. I still have one issue that the final exported STEP file shows a surface body and a solidbody, What I would like is a single solid body with 7 surfaces instead of the original 6.

Sharing script here:
```python
import cadquery as cq
from jupyter_cadquery import show
from OCP.BRepFeat import BRepFeat_SplitShape
from cadquery.occ_impl.shapes import Shape

# Step 1: Define the base model (the box)
base_model = cq.Workplane("XY").box(10, 10, 10, centered=True)

# show(base_model)

# Step 2: Define the wire on the top face as a triangle
# We create a triangular wire by connecting three points on the top face

top_wire = (
    base_model.faces(">Z")  # Select the top face
    .workplane()             # Go to the local workplane of the top face
    .moveTo(-2.5, -2.5)          # Start at one corner of the top face
    .lineTo(2.5, -2.5)           # Draw the first line to the second point
    .lineTo(0, 2.5)            # Draw the second line to the third point
    .close()                 # Close the triangle by connecting back to the starting point
    .wire()                  # Create a wire from the closed triangle
)

show(base_model, top_wire)

# Step 3: Extract the wire as a TopoDS_Shape
# The `wrapped` attribute converts the CadQuery object to a TopoDS_Shape
split_wire_shape = top_wire.val().wrapped

print(f'split_wire_shape is of type {type(split_wire_shape)}')

# Step 4: Initialize the BRepFeat_SplitShape with the top face shape
top_face_shape = base_model.faces(">Z").val().wrapped
split_shape = BRepFeat_SplitShape(top_face_shape)

print(f'split_shape is of type {type(split_shape)}')

# Step 5: Add the wire to split the shape
split_shape.Add(split_wire_shape, top_face_shape)

# Step 6: Build the split operation
split_shape.Build()

# Step 7: Check if the operation was successful and get the resulting shape
if split_shape.IsDone():
    result_shape = split_shape.Shape()
    print("Splitting operation completed successfully.")
else:
    print("Splitting operation failed.")

# show(result_shape, base_model)

# Step 1: Convert `result_shape` to a CadQuery-compatible Shell or Solid
result_cq_shape = Shape.cast(result_shape)

# Step 2: Add result_cq_shape to base_model
combined_model = base_model.add(result_cq_shape)

# Display the combined model
show(combined_model)

```

I'm unable to attach a STEP file so here is the screenshot.
![image](https://github.com/user-attachments/assets/1badd6c1-d55d-4b73-8452-c129ff9b16aa)

Any advice to point me in the right direction is appreciated!

PS: script is generated with lot of chatgpt help

**Reply by adam-urbanczyk** [reply]
Did you look at https://github.com/CadQuery/cadquery/pull/1353 as suggested above?

**Reply by pruj-speed** [reply]
@adam-urbanczyk I did look at the imprint function that already exists under assembly. If I understand correctly, this function works on Solids. The ask here is to split a surface into multiple surfaces. Are you saying I can achieve this with the imprint already? Would appreciate an example if that's the case.

**Reply by adam-urbanczyk** [reply]
```python
from cadquery.occ_impl.shapes import *
from cadquery.vis import show

b = box(1,1,1)
c = face(circle(0.1)).moved(z=1)

res = b/c # same as split(b, c)

show(res) # for viewing
```
NB: using master.

**Reply by pruj-speed** [reply]
@adam-urbanczyk Thank you for pointing out the split() method. Did not know it already existed. I now can split a face.

Sharing the script for anyone who might want to do this in the future.

```python
from cadquery.occ_impl.shapes import polyline, face
from cadquery.vis import show
import cadquery as cq

#define points that lie on a triangle
points = [
    (-2.5, -2.5),
    (2.5, -2.5),
    (0, 2.5)
]

# Step 1: Create the base cube
base_model = cq.Workplane("XY").box(10, 10, 10, centered=True)

# Step 2: Create a closed wire using the polyline function
polygon_wire = polyline(*points).close()

# Step 3: Convert the wire to a face and move to box top
polygon_face = face(polygon_wire).moved(z=5)

# Step 5: Perform the split operation
result = base_model.split(polygon_face)

# Display the result for visualization
show(result)
```
Now the split face can be picked in a simulation tool for assigning a boundary condition.
![image](https://github.com/user-attachments/assets/3446c629-623a-41b6-bb11-77bd22e80b70)

@fsfeller Tagging you incase this is what you were looking for
